packages/transactions/src/keys.ts

```diff
@@ -108,19 +108,32 @@ export function serializePublicKey(key: StacksPublicKey): Buffer {
   return bufferArray.concatBuffer();
 }
 
+export function isPrivateKeyCompressed(key: string | Uint8Array) {
+  const data = typeof key === 'string' ? hexToBytes(key) : key;
+  let compressed = false;
+  if (data.length === 33) {
+    if (data[data.length - 1] !== 1) {
+      throw new Error(
+        'Improperly formatted private-key. 33 byte length usually ' +
+          'indicates compressed key, but last byte must be == 0x01'
+      );
+    }
+    compressed = true;
+  }
+  return compressed;
+}
```

Comment:
Did the library previously default to compressed or default to uncompressed?

Reply:
Nevermind -- I see this is the correct behavior. Can you update `createStacksPrivateKey` to invoke this method rather than duplicate the logic?